### PR TITLE
1284

### PR DESCRIPTION
**Pull request by m-kovalsky**
#1284

### CODE REVIEW

**Comment by m-kovalsky** [comment]
#1111

File: src/sempy_labs/_generate_semantic_model.py [hunk 1]
```diff
@@ -630,35 +630,26 @@ def get_semantic_model_size(
...
-    elif model_size >= 10**6:
-        result = model_size / (1024**2) * 10**6
-    elif model_size >= 10**3:
-        result = model_size / (1024) * 10**3
-    else:
-        result = model_size
+    result = fabric.evaluate_dax(dataset=dataset, workspace=workspace, dax_string=dax)
 
-    return float(result)
+    return int(float(result.iloc[0, 0]))
```
Comment (greatvovan) [inline src/sempy_labs/_generate_semantic_model.py:655]: Just curious, why `int(float(...))`? What's the original type returned by `result.iloc[0, 0]`?

**Comment by greatvovan** [comment]
For awareness: https://github.com/microsoft/semantic-link-labs/issues/1284#issuecomment-5029104389

File: src/sempy_labs/_generate_semantic_model.py (hunk 1 above)
Comment (m-kovalsky) [reply src/sempy_labs/_generate_semantic_model.py:655]: when in a pyspark notebook, this caused issues unless i cast it as a float first.

File: src/sempy_labs/lakehouse/_lakehouse.py [hunk 2]
```diff
@@ -256,11 +233,17 @@ def vacuum_lakehouse_tables(
...
         full_table_name = f"{schema_name}.{table_name}"
-        path = r["Location"]
         bar.set_description(
             f"Vacuuming the '{full_table_name if schema_name else table_name}' table ({idx + 1}/{total})..."
         )
-        _vacuum_table(path=path, retain_n_hours=retain_n_hours)
+        run_table_maintenance(
+            table_name=table_name,
+            vacuum=True,
+            retention_period=f"{retain_n_hours // 24}:{retain_n_hours % 24}:00:00"
```
Comment (whiskyboy) [inline src/sempy_labs/lakehouse/_lakehouse.py]: missing a "," here?

**Review by whiskyboy — APPROVED** [review]

**Review by whiskyboy — APPROVED** [review]

File: src/sempy_labs/_generate_semantic_model.py (hunk 1 above)
Comment (greatvovan) [reply src/sempy_labs/_generate_semantic_model.py:655]: My tests show that the original type is `numpy.int64`.  This should not cause issues converting into Python `int` (in fact, it can hold more). Did you investigate it? Maybe it was some edge case conversion of `NaN` or something like that, which we should handle?
